crates/lib/src/db/raw_def/v9.rs

```diff
@@ -0,0 +1,467 @@
+//! This module contains the "raw definitions" returned by modules to the database.
+
+use crate::db::auth::{StAccess, StTableType};
+use itertools::Itertools;
+use spacetimedb_primitives::*;
+use spacetimedb_sats::AlgebraicType;
+use spacetimedb_sats::AlgebraicTypeRef;
+use spacetimedb_sats::ProductTypeElement;
+use spacetimedb_sats::{de, ser, Typespace};
+
+/// A not-yet-validated identifier.
+pub type RawIdentifier = Box<str>;
+
+/// A possibly-invalid raw module definition.
+///
+/// ABI Version 9.
+///
+/// These "raw definitions" may contain invalid data, and are validated by the `validate` module into a proper `spacetimedb_schema::ModuleDef`, or a collection of errors.
+///
+/// The module definition has a single logical global namespace, which maps `Identifier`s to:
+///
+/// - database-level objects:
+///     - logical schema objects:
+///         - tables
+///         - constraints
+///         - sequence definitions
+///     - physical schema objects:
+///         - indexes
+/// - module-level objects:
+///     - reducers
+///     - schedule definitions
+/// - binding-level objects:
+///     - type aliases
+///
+/// All of these types of objects must have unique names within the module.
+/// The exception is columns, which need unique names only within a table.
+#[derive(Debug, Clone, Default, ser::Serialize, de::Deserialize)]
+#[cfg_attr(feature = "test", derive(PartialEq, Eq, PartialOrd, Ord))]
+pub struct RawModuleDefV9 {
+    /// The types used in the module.
+    ///
+    /// `AlgebraicTypeRef`s in the table, reducer, and type alias declarations refer to this typespace.
+    ///
+    /// Any `Product` or `Sum` types used transitively by the module MUST be declared in this typespace.
+    /// Each should have EXACTLY ONE corresponding `TypeAlias` declaration in the `misc_exports` field,
+    /// with a module-unique name.
+    /// This ensures that client code generation will work correctly.
+    ///
+    /// It is permitted but not required to refer to `Builtin` types via this typespace.
```

Comment:
```suggestion
    /// It is permitted but not required to refer to `Builtin` or "primitive" types via this typespace.
```
We're eventually removing the `Builtin` distinction so it'll be a little confusing